### PR TITLE
🏷 Tag abstract parts in JATS

### PR DESCRIPTION
**Pull request by rowanc1**
This adds tagging of abstract parts in JATS. Right now the `plain-language summary` and `keypoints` are added by default.

### CODE REVIEW

File: packages/myst-common/src/extractParts.ts [hunk 1]
```diff
@@ -38,6 +54,7 @@ export function extractPart(tree: GenericParent, part: string): GenericParent |
         delete block.data.tags;
       }
     }
+    if (opts?.removePartData) delete block.data.part;
```
Comment (fwkoch) [inline packages/myst-common/src/extractParts.ts:57]: This doesn't delete the part data if it's in `tags` - is that an edge case we should keep consistent?

File: packages/myst-to-jats/src/index.ts [hunk 2]
```diff
@@ -520,23 +561,29 @@ class JatsSerializer implements IJatsSerializer {
...
-          isNotebookArticleRep: this.data.isNotebookArticleRep,
-          slug: this.data.slug,
-          handlers: this.handlers,
-        });
-        abstractSerializer.render();
-        this.data.abstract = abstractSerializer.elements();
-      }
+      const abstractParts = opts.abstractParts ?? [{ part: 'abstract' }];
+      this.data.abstracts = abstractParts
+        .map((def) => renderAbstract(this.file, this.mdast, def, opts))
```
Comment (fwkoch) [inline packages/myst-to-jats/src/index.ts:566]: This seems fine, we are doing slightly more pre-rendering here, then just grab the outputs later. More consolidated than what we were doing with `getAbstract` before. 👍

File: packages/myst-to-jats/src/index.ts [hunk 3]
```diff
@@ -520,23 +561,29 @@ class JatsSerializer implements IJatsSerializer {
...
+        .filter((e) => !!e) as Element[];
     }
+    this.data.acknowledgments = renderAcknowledgments(this.file, this.mdast, opts);
+    const backSections = opts?.backSections ?? [
+      {
+        part: ['data-availability', 'data_availability', 'availability'],
```
Comment (fwkoch) [inline packages/myst-to-jats/src/index.ts:572]: Should this just go in `myst-cli` too, with the various abstract sections? Thinking ahead to making this more extensible, seems like we should have the "myst defaults" in only one place (either here or myst-cli).

**Review by fwkoch — APPROVED** [review]
Nice, this looks like a good start, lifting out some jats extensibility.

File: packages/myst-common/src/extractParts.ts (hunk 1 above)
Comment (rowanc1) [reply packages/myst-common/src/extractParts.ts:57]: Hm, yeah, this needs to be fixed. Will add a test. Thanks.

File: packages/myst-to-jats/src/index.ts (hunk 3 above)
Comment (rowanc1) [reply packages/myst-to-jats/src/index.ts:572]: Yep, good point -- I will move it up to the CLI for now.

**Review by rowanc1 — COMMENTED** [review]
Thanks for the review -- will get a few of those changes completed now.

**Pull request merged**